cmd/gardenlet/app/app.go

```
@@ -460,6 +467,30 @@ func (g *garden) cleanupOrphanedExtensionsServiceAccounts(ctx context.Context, g
 	return flow.Parallel(taskFns...)(ctx)
 }
 
+// TODO(Kostov6): Remove this code after v1.89 has been released.
+func (g *garden) cleanupGRMSecretFinalizers(ctx context.Context, seedClient client.Client, log logr.Logger) error {
+	mrs := &resourcesv1alpha1.ManagedResourceList{}
+	if err := seedClient.List(ctx, mrs); err != nil {
+		log.Error(err, "Failed to list ManagedResources while cleaing up GRM finalizers")
```

return error instead of logging